### ISSUE TITLE
Unnecessary escape characters in regular expression

### ISSUE THREAD

**Issue by Michael-1** [browserslist/browserslist-useragent-regexp#472]
In the generated regular expression, dot characters in ranges are unnecessarily escaped. `[_\.]` can be written simply as `[_.]`, as the dot character is always interpreted literally  in a range `[]`.

**Comment by elliottregan** [comment]
Our ESLint rules picked this up, as well. Minor change, but every byte helps.

**Comment by dangreen** [comment]
@Michael-1 @elliottregan Hi! Can you provide example for reproduction ?

**Comment by elliottregan** [comment]
@dangreen This is a generated file that gets flagged by [ESLint's rule](https://eslint.org/docs/rules/no-useless-escape): 
```js
module.exports = /((CPU[ +]OS|iPhone[ +]OS|CPU[ +]iPhone|CPU IPhone OS)[ +]+(12[_\.]2|12[_\.]([3-9]|\d{2,})|12[_\.]4|12[_\.]([5-9]|\d{2,})|(1[3-9]|[2-9]\d|\d{3,})[_\.]\d+|13[_\.]0|13[_\.]([1-9]|\d{2,})|13[_\.]3|13[_\.]([4-9]|\d{2,})|(1[4-9]|[2-9]\d|\d{3,})[_\.]\d+)(?:[_\.]\d+)?)|(SamsungBrowser\/(10\.1|10\.([2-9]|\d{2,})|(1[1-9]|[2-9]\d|\d{3,})\.\d+))|((Chromium|Chrome)\/(79\.0|79\.([1-9]|\d{2,})|([8-9]\d|\d{3,})\.\d+)(?:\.\d+)?)|(Version\/(13\.0|13\.([1-9]|\d{2,})|(1[4-9]|[2-9]\d|\d{3,})\.\d+)(?:\.\d+)? Safari\/)|(Firefox\/(72\.0|72\.([1-9]|\d{2,})|(7[3-9]|[8-9]\d|\d{3,})\.\d+)\.\d+)|(Firefox\/(72\.0|72\.([1-9]|\d{2,})|(7[3-9]|[8-9]\d|\d{3,})\.\d+)(pre|[ab]\d+[a-z]*)?)/;
```
[Demo](https://eslint.org/demo#eyJ0ZXh0IjoibW9kdWxlLmV4cG9ydHMgPSAvKChDUFVbICtdT1N8aVBob25lWyArXU9TfENQVVsgK11pUGhvbmV8Q1BVIElQaG9uZSBPUylbICtdKygxMltfXFwuXTJ8MTJbX1xcLl0oWzMtOV18XFxkezIsfSl8MTJbX1xcLl00fDEyW19cXC5dKFs1LTldfFxcZHsyLH0pfCgxWzMtOV18WzItOV1cXGR8XFxkezMsfSlbX1xcLl1cXGQrfDEzW19cXC5dMHwxM1tfXFwuXShbMS05XXxcXGR7Mix9KXwxM1tfXFwuXTN8MTNbX1xcLl0oWzQtOV18XFxkezIsfSl8KDFbNC05XXxbMi05XVxcZHxcXGR7Myx9KVtfXFwuXVxcZCspKD86W19cXC5dXFxkKyk/KXwoU2Ftc3VuZ0Jyb3dzZXJcXC8oMTBcXC4xfDEwXFwuKFsyLTldfFxcZHsyLH0pfCgxWzEtOV18WzItOV1cXGR8XFxkezMsfSlcXC5cXGQrKSl8KChDaHJvbWl1bXxDaHJvbWUpXFwvKDc5XFwuMHw3OVxcLihbMS05XXxcXGR7Mix9KXwoWzgtOV1cXGR8XFxkezMsfSlcXC5cXGQrKSg/OlxcLlxcZCspPyl8KFZlcnNpb25cXC8oMTNcXC4wfDEzXFwuKFsxLTldfFxcZHsyLH0pfCgxWzQtOV18WzItOV1cXGR8XFxkezMsfSlcXC5cXGQrKSg/OlxcLlxcZCspPyBTYWZhcmlcXC8pfChGaXJlZm94XFwvKDcyXFwuMHw3MlxcLihbMS05XXxcXGR7Mix9KXwoN1szLTldfFs4LTldXFxkfFxcZHszLH0pXFwuXFxkKylcXC5cXGQrKXwoRmlyZWZveFxcLyg3MlxcLjB8NzJcXC4oWzEtOV18XFxkezIsfSl8KDdbMy05XXxbOC05XVxcZHxcXGR7Myx9KVxcLlxcZCspKHByZXxbYWJdXFxkK1thLXpdKik/KS87Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjoyLCJmb3ItZGlyZWN0aW9uIjoyLCJnZXR0ZXItcmV0dXJuIjoyLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjoyLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6Miwibm8tY2xhc3MtYXNzaWduIjoyLCJuby1jb21wYXJlLW5lZy16ZXJvIjoyLCJuby1jb25kLWFzc2lnbiI6Miwibm8tY29uc3QtYXNzaWduIjoyLCJuby1jb25zdGFudC1jb25kaXRpb24iOjIsIm5vLWNvbnRyb2wtcmVnZXgiOjIsIm5vLWRlYnVnZ2VyIjoyLCJuby1kZWxldGUtdmFyIjoyLCJuby1kdXBlLWFyZ3MiOjIsIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6Miwibm8tZHVwZS1lbHNlLWlmIjoyLCJuby1kdXBlLWtleXMiOjIsIm5vLWR1cGxpY2F0ZS1jYXNlIjoyLCJuby1lbXB0eSI6Miwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1lbXB0eS1wYXR0ZXJuIjoyLCJuby1leC1hc3NpZ24iOjIsIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6Miwibm8tZXh0cmEtc2VtaSI6Miwibm8tZmFsbHRocm91Z2giOjIsIm5vLWZ1bmMtYXNzaWduIjoyLCJuby1nbG9iYWwtYXNzaWduIjoyLCJuby1pbXBvcnQtYXNzaWduIjoyLCJuby1pbm5lci1kZWNsYXJhdGlvbnMiOjIsIm5vLWludmFsaWQtcmVnZXhwIjoyLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6Miwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6Miwibm8tbmV3LXN5bWJvbCI6Miwibm8tb2JqLWNhbGxzIjoyLCJuby1vY3RhbCI6Miwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjoyLCJuby1yZWRlY2xhcmUiOjIsIm5vLXJlZ2V4LXNwYWNlcyI6Miwibm8tc2VsZi1hc3NpZ24iOjIsIm5vLXNldHRlci1yZXR1cm4iOjIsIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjoyLCJuby1zcGFyc2UtYXJyYXlzIjoyLCJuby10aGlzLWJlZm9yZS1zdXBlciI6Miwibm8tdW5kZWYiOjIsIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjoyLCJuby11bnJlYWNoYWJsZSI6Miwibm8tdW5zYWZlLWZpbmFsbHkiOjIsIm5vLXVuc2FmZS1uZWdhdGlvbiI6Miwibm8tdW51c2VkLWxhYmVscyI6Miwibm8tdW51c2VkLXZhcnMiOjIsIm5vLXVzZWxlc3MtY2F0Y2giOjIsIm5vLXVzZWxlc3MtZXNjYXBlIjoyLCJuby13aXRoIjoyLCJyZXF1aXJlLXlpZWxkIjoyLCJ1c2UtaXNuYW4iOjIsInZhbGlkLXR5cGVvZiI6Mn0sImVudiI6e319fQ==)

**Comment by dangreen** [comment]
@elliottregan @Michael-1 fixed in [v2.1.1](https://github.com/browserslist/browserslist-useragent-regexp/releases/tag/v2.1.1)

**Comment by KrzysztofWidackiPracuj** [comment]
On version 2.1.1 there are still some unnecessary characters.

My browserlistrc 
last 2 versions
not dead
not ie >= 0

eslint error:
`Unnecessary escape character: \/  no-useless-escape`

```
/((CPU[ +]OS|iPhone[ +]OS|CPU[ +]iPhone|CPU IPhone OS)[ +]+(13[_.]4|13[_.]([5-9]|\d{2,})|13[_.]7|13[_.]([8-9]|\d{2,})|(1[4-9]|[2-9]\d|\d{3,})[_.]\d+|14[_.]0|14[_.]([1-9]|\d{2,})|14[_.]2|14[_.]([3-9]|\d{2,})|(1[5-9]|[2-9]\d|\d{3,})[_.]\d+)(?:[_.]\d+)?)|(Opera Mini(?:\/att)?\/?(\d+)?(?:\.\d+)?(?:\.\d+)?)|(Opera\/.+Opera Mobi.+Version\/(59\.0|59\.([1-9]|\d{2,})|([6-9]\d|\d{3,})\.\d+))|(Opera\/(59\.0|59\.([1-9]|\d{2,})|([6-9]\d|\d{3,})\.\d+).+Opera Mobi)|(Opera Mobi.+Opera(?:\/|\s+)(59\.0|59\.([1-9]|\d{2,})|([6-9]\d|\d{3,})\.\d+))|(SamsungBrowser\/(12\.0|12\.([1-9]|\d{2,})|(1[3-9]|[2-9]\d|\d{3,})\.\d+))|(Edge\/(86(?:\.0)?|86(?:\.([1-9]|\d{2,}))?|(8[7-9]|9\d|\d{3,})(?:\.\d+)?))|((Chromium|Chrome)\/(86\.0|86\.([1-9]|\d{2,})|(8[7-9]|9\d|\d{3,})\.\d+)(?:\.\d+)?)|(IEMobile[ \/](11\.0|11\.([1-9]|\d{2,})|(1[2-9]|[2-9]\d|\d{3,})\.\d+))|(Version\/(13\.1|13\.([2-9]|\d{2,})|(1[4-9]|[2-9]\d|\d{3,})\.\d+|14\.0|14\.([1-9]|\d{2,})|(1[5-9]|[2-9]\d|\d{3,})\.\d+)(?:\.\d+)? Safari\/)|(Firefox\/(82\.0|82\.([1-9]|\d{2,})|(8[3-9]|9\d|\d{3,})\.\d+)\.\d+)|(Firefox\/(82\.0|82\.([1-9]|\d{2,})|(8[3-9]|9\d|\d{3,})\.\d+)(pre|[ab]\d+[a-z]*)?)/

```

**Comment by dangreen** [comment]
@KrzysztofWidackiPracuj Hi. Please try [v4.0.0-beta.1](https://github.com/browserslist/browserslist-useragent-regexp/releases/tag/v4.0.0-beta.1).

**Comment by dargmuesli** [comment]
Using `defaults` `browserslist-useragent-regexp --allowHigherVersions` returns:
```
/Edge?\/(10[7-9]|1[1-9]\d|[2-9]\d{2}|\d{4,})(\.\d+|)(\.\d+|)|Firefox\/(10[2-9]|1[1-9]\d|[2-9]\d{2}|\d{4,})\.\d+(\.\d+|)|Chrom(ium|e)\/(10[6-9]|1[1-9]\d|[2-9]\d{2}|\d{4,})\.\d+(\.\d+|)|Maci.* Version\/(15\.([6-9]|\d{2,})|(1[6-9]|[2-9]\d|\d{3,})\.\d+)([,.]\d+|)( Mobile\/\w+|) Safari\/|Chrome.+OPR\/(9[1-9]|\d{3,})\.\d+\.\d+|(CPU[ +]OS|iPhone[ +]OS|CPU[ +]iPhone|CPU IPhone OS|CPU iPad OS)[ +]+(14[._]([5-9]|\d{2,})|(1[5-9]|[2-9]\d|\d{3,})[._]\d+)([._]\d+|)|Opera Mini|Android:?[ /\-](10[89]|1[1-9]\d|[2-9]\d{2}|\d{4,})(\.\d+|)(\.\d+|)|Mobile Safari.+OPR\/(7[2-9]|[89]\d|\d{3,})\.\d+\.\d+|Android.+Firefox\/(10[7-9]|1[1-9]\d|[2-9]\d{2}|\d{4,})\.\d+(\.\d+|)|Android.+Chrom(ium|e)\/(10[89]|1[1-9]\d|[2-9]\d{2}|\d{4,})\.\d+(\.\d+|)|Android.+(UC? ?Browser|UCWEB|U3)[ /]?(13\.([4-9]|\d{2,})|(1[4-9]|[2-9]\d|\d{3,})\.\d+)\.\d+|SamsungBrowser\/(1[89]|[2-9]\d|\d{3,})\.\d+|Android.+MQ{2}Browser\/(13(\.([1-9]|\d{2,})|)|(1[4-9]|[2-9]\d|\d{3,})(\.\d+|))(\.\d+|)|K[Aa][Ii]OS\/(2\.([5-9]|\d{2,})|([3-9]|\d{2,})\.\d+)(\.\d+|)/
```

which includes `[ /\-]` and thus an `Unnecessary escape character: \-` at `1:495`. I'm using
- "browserslist": "4.21.4"
- "browserslist-useragent-regexp": "4.0.0"

so @dangreen, can this issue be reopened for this to be fixed or shall I create a new ticket?

**Comment by dangreen** [comment]
@dargmuesli Hi. Fixed in [ua-regexes-lite@1.1.4](https://github.com/TrigenSoftware/ua-regexes-lite/releases/tag/v1.1.4). Reinstall browserslist-useragent-regexp to trigger ua-regexes-lite update, or just install it as dev dependency.

**Comment by dargmuesli** [comment]
Great! Thank you for the quick fix :)
